ros_gz_sim/src/create.cpp

```diff
@@ -30,54 +30,51 @@
 #include <std_msgs/msg/string.hpp>
 
 
-DEFINE_string(world, "", "World name.");
-DEFINE_string(file, "", "Load XML from a file.");
-DEFINE_string(param, "", "Load XML from a ROS param.");
-DEFINE_string(string, "", "Load XML from a string.");
-DEFINE_string(topic, "", "Load XML from a ROS string publisher.");
-DEFINE_string(name, "", "Name for spawned entity.");
-DEFINE_bool(allow_renaming, false, "Rename entity if name already used.");
-DEFINE_double(x, 0, "X component of initial position, in meters.");
-DEFINE_double(y, 0, "Y component of initial position, in meters.");
-DEFINE_double(z, 0, "Z component of initial position, in meters.");
-DEFINE_double(R, 0, "Roll component of initial orientation, in radians.");
-DEFINE_double(P, 0, "Pitch component of initial orientation, in radians.");
-DEFINE_double(Y, 0, "Yaw component of initial orientation, in radians.");
-
 // ROS interface for spawning entities into Gazebo.
 // Suggested for use with roslaunch and loading entities from ROS param.
 // If these are not needed, just use the `gz service` command line instead.
 int main(int _argc, char ** _argv)
 {
-  auto filtered_arguments = rclcpp::init_and_remove_ros_arguments(_argc, _argv);
+   rclcpp::init(_argc, _argv);
   auto ros2_node = rclcpp::Node::make_shared("ros_gz_sim");
 
-  // Construct a new argc/argv pair from the flags that weren't parsed by ROS
-  // Gflags wants a mutable pointer to argv, which is why we can't use a
-  // vector of strings here
-  int filtered_argc = filtered_arguments.size();
-  char ** filtered_argv = new char *[(filtered_argc + 1)];
-  for (int ii = 0; ii < filtered_argc; ++ii) {
-    filtered_argv[ii] = new char[filtered_arguments[ii].size() + 1];
-    snprintf(filtered_argv[ii], sizeof(filtered_argv), "%s", filtered_arguments[ii].c_str());
-  }
-  filtered_argv[filtered_argc] = nullptr;
-
-  gflags::AllowCommandLineReparsing();
-  gflags::SetUsageMessage(
-    R"(Usage: create -world [arg] [-file FILE] [-param PARAM] [-topic TOPIC]
-                       [-string STRING] [-name NAME] [-allow_renaming RENAMING] [-x X] [-y Y] [-z Z]
-                       [-R ROLL] [-P PITCH] [-Y YAW])");
-  gflags::ParseCommandLineFlags(&filtered_argc, &filtered_argv, false);
-
-  // Free our temporary argc/argv pair
-  for (size_t ii = 0; filtered_argv[ii] != nullptr; ++ii) {
-    delete[] filtered_argv[ii];
-  }
-  delete[] filtered_argv;
-
+  std::string world_,file_,param_,string_,topic_,name_;
+  bool allow_renaming_;
+  double x_,y_,z_,R_,P_,Y_;
```

Comment:
In general we use underscore at the end when the variables are attributes of class, do you mind to remove the undercore at the end of tha variable name ?

```suggestion
  std::string world_, file_, param_, string_, topic_, name_;
  bool allow_renaming_;
  double x_, y_, z_, R_, P_, Y_;
```

Reply:
Hi @ahcorde , I have made the changes could you  please review again. Thank you.